packages/beacon-node/src/api/impl/validator/index.ts

```diff
@@ -525,15 +532,18 @@ export function getValidatorApi({
 
     const builderPayloadValue = blindedBlock?.executionPayloadValue ?? BigInt(0);
     const enginePayloadValue = fullBlock?.executionPayloadValue ?? BigInt(0);
+    const consensusBlockValueBuilder = blindedBlock?.consensusBlockValue ?? BigInt(0);
+    const consensusBlockValueEngine = fullBlock?.consensusBlockValue ?? BigInt(0);
+
+    const blockValueBuilder = builderPayloadValue + consensusBlockValueBuilder;
+    const blockValueEngine = enginePayloadValue + consensusBlockValueEngine;
 
     let selectedSource: ProducedBlockSource | null = null;
 
     if (fullBlock && blindedBlock) {
       switch (builderSelection) {
         case routes.validator.BuilderSelection.MaxProfit: {
-          // If executionPayloadValues are zero, than choose builder as most likely beacon didn't provide executionPayloadValue
-          // and builder blocks are most likely thresholded by a min bid
-          if (enginePayloadValue >= builderPayloadValue && enginePayloadValue !== BigInt(0)) {
+          if (blockValueEngine > blockValueBuilder) {
```

Comment:
@g11tech Can you confirm if this statement looks good to you? I think we choose `engine` only if its block value is greater than builder’s block value. Not sure if it matters which one to choose if they are equal. 
Also no more zero check on the payload value.